src/EnergyPlus/GroundTemperatureModeling/FiniteDifferenceGroundTemperatureModel.cc

```diff
@@ -301,7 +301,7 @@ void FiniteDiffGroundTempsModel::getWeatherData(EnergyPlusData &state)
                 airDensity_num += state.dataEnvrn->OutAirDensity;
                 relHum_num += state.dataEnvrn->OutRelHumValue;
                 windSpeed_num += state.dataEnvrn->WindSpeed;
-                horizSolarRad_num += state.dataEnvrn->BeamSolarRad + state.dataEnvrn->DifSolarRad;
+                horizSolarRad_num += max(state.dataEnvrn->SOLCOS(3), 0.0) * state.dataEnvrn->BeamSolarRad + state.dataEnvrn->DifSolarRad;
```

Comment:
:+1:

Reply:
This does not need to be part of this PR, but put a pin in it anyway. Change SOLCOS from `Array1D<Real64>(3)` to `Vector3<Real64>` and access the components by `.x`, `.y`, and `.z` rather than by index. It's both faster (probably does not matter that much) and more readable (IMO).

Reply:
Agreed, that is definitely a better structure to use here.

Reply:
@amirroth So, do you feel that this this important enough to make into its own defect or do you feel like it can wait until the next time someone works on solar radiation (no idea when that might be)?

Reply:
Should not let this slide indefinitely, so please make a separate defect. `SOLCOS` is not the example of this particular idiom. It would be good to do a larger sweep and convert all of them. Thx.

Reply:
Got it.  New defect is now in the system.  See: https://github.com/NREL/EnergyPlus/issues/8893